proxy/src/outbound.rs

```diff
@@ -54,6 +54,60 @@ impl<B> Outbound<B> {
             bind_timeout,
         }
     }
+
+    const DEFAULT_PORT: Option<u16> = Some(80);
+
+    /// TODO: Return error when `HostAndPort::normalize()` fails.
```

Comment:
If we're discarding the error here, maybe we should at least `.inspect_err(|e| error!("normalizing {:?} failed: {:?}", authority, e))` first (or something like that)?

 I know the original code you're refactoring didn't do that, but it seems simple enough that we could add it in this PR...

Reply:
What does that do?

Reply:
Log the error prior to discarding it.

Reply:
Is this part of the std lib or is it an extension? I don't see this defined on `Result`. In any case, let's address this separately.

Reply:
> Is this part of the std lib or is it an extension? 

Ah, nevermind, that's a method on `Stream`, I think, that I mistakenly assumed was also on `Result`s; `map_err` would do more or less the same thing here though, since the error value gets discarded anyway.

> In any case, let's address this separately.
👍